src/components/SignUp.js

```diff
@@ -1,74 +1,164 @@
+import * as Yup from 'yup';
+import { useNavigate } from "react-router-dom";
+
 
 export default function SignUp() {
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [gender, setGender] = useState('');
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [repeatPassword, setRepeatPassword] = useState('');
+    const [error, setError] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+
+    let navigate = useNavigate(); 
+    
+    const signupSchema = Yup.object().shape({
+        firstName: Yup.string().max(15).required(),
+        lastName: Yup.string().max(15).required(),
+        username: Yup.string().max(15).required(),
+        email: Yup.string().email().required(),
+        password: Yup.string().min(8).max(20).required(),
+        repeatPassword: Yup.string()
+            .oneOf([Yup.ref('password'), null])
+            .required(),
+    });
+
+    
+
+    const handleClick = async (e) => {
+        e.preventDefault();
+        setError(false); 
+        setErrorMessage(''); 
+        const isValid = await signupSchema.isValid({
+            firstName: firstName,
+            lastName: lastName,
+            username: username,
+            email: email,
+            password: password,
+            repeatPassword: repeatPassword,
+        });
+
+        if (isValid) {
+            axiosInstance
+                .post('/auth/signup', {
+                    firstName: firstName,
+                    lastName: lastName,
+                    username: username,
+                    email: email,
+                    password: password,
+                    gender: gender,
+                })
+                .then((response) => {
+                    alert("Account has been created!");
+                    navigate('/auth/signin');
+                    console.log(response);
+                })
+                .catch((err) => {
+                    console.log(err);
+                    console.log(err.response);
+                    setErrorMessage(err.response.data.msg); 
+                    setError(true);
+                });
+        } else {
+            setError(true); 
+            setErrorMessage('One or more input is invalid');
+        }
+    };
+
     return (
         <Container id="main-container" className="d-grid h-100">
-            <Row>
-                <Col sm={7}>
-                    <img class="info-Banner" src={picLogScreen} alt="picLogScreen"/>
-                </Col>
-
-                <Col sm={4}>
+            
+            <div className = "split left">
+                <div className="centered">
+                    <img className="info-Banner" src={picLogScreen} alt="picLogScreen"/>
+                </div>
+            </div>
+              
+            <div className = "split right">
```

Comment:
Class names are typically written as `split-right`

Reply:
Different css for both, so when I work on the css, its not split-right, its split and right seperately.